client/asset/eth/nodeclient.go

```diff
@@ -207,20 +202,32 @@ func (n *nodeClient) balance(ctx context.Context) (*Balance, error) {
 	}
 
 	outgoing := new(big.Int)
+	incoming := new(big.Int)
+	zero := new(big.Int)
 	for _, tx := range pendingTxs {
 		// TODO: Check that the from address matches addr. types.Transaction
 		// does not expose the from address directly.
-		outgoing.Add(outgoing, tx.Value())
+		gas := new(big.Int).SetUint64(tx.Gas())
+		fees := new(big.Int).Mul(gas, tx.GasFeeCap())
+		outgoing.Add(outgoing, fees)
+		v := tx.Value()
+		if v.Cmp(zero) == 0 {
+			for ver, c := range n.contractors {
+				in, err := c.incomingValue(ctx, tx)
+				if err != nil {
+					n.log.Errorf("version %d contractor incomingValue error: %v", ver, err)
```

Comment:
and `continue`?

Comments on each side of the `if v.Cmp(zero) == 0 {` conditional would be nice too I think (e.g. if sending non-zero value add to outgoing; if zero value, attempt to detect contract interaction that pays to the account).